Network foundation

haydencheers opened the pull request:
epic winz

haydencheers committed:
Initial commit to lattice/foundation.
Has ability to broadcast service.

haydencheers committed:
Remoting Channel Implementation

haydencheers committed:
Minor Changes

haydencheers committed:
Basic automated sharing of text.

haydencheers committed:
Resolved security exception when sending images by changing security level to allow images. Might be better to send image as a serialised string or something?

haydencheers committed:
Refactored code into seperate files.
Refactored discovery code, now focused only on service discovery

haydencheers committed:
Changes in recoding services to stop exceptions.
Created basic command line tool to test Lattice.

twisty-n commented:
@haydencheers @alistairwoodcock 
Did you want me to do the rebase and deal with merge conflicts when this has been reviewed?

haydencheers committed:
Modified something to stop exceptions.

haydencheers committed:
Converted to WCF. Remoting would cause odd behaviour to occur. Images can be send accross network.

haydencheers committed:
Major refactoring and redesign of appliaction. Fleet code seperated into library. Remoted remoting code, everything is WCF.
There are three applications for using image transfer:
1. Fleet Daemon. A command line application to boradcast the service.
2. Fleet.Test. The command line application, used to send files from a remote machine. A GUI may be built.
3. FleetViewer. A  WInforms based application to open the received images sent to Fleet Daemon.

FleetDaemon and FleetViewer communicate via WCF pipes.
FleetDaemon and Fleet.Test communication via WCF TCP sockets.

I can confirm that this works between two windows 10 machines on my local network. Not confirmed if this works on Mono.
Fleet service name bradcast is not implemented, all broadcasts will use the service name Fleet. This is not a huge issue, but will be rectified in a later update.

haydencheers committed:
Basic image sharing app. Not pretty. Successfuly shares images accross network.

haydencheers committed:
Basic image sharing app. Not pretty. Successfuly shares images accross network.

twisty-n committed:
Add some references and clean up base project
